sharing/present.go

```diff
@@ -116,44 +120,51 @@ func (handler *Handler) showObject(ctx context.Context, w http.ResponseWriter, r
 		return nil
 	}
 
-	ipBytes, err := object.GetObjectIPs(ctx, uplink.Config{}, pr.access, pr.bucket, o.Key)
-	if err != nil {
-		return WithAction(err, "get object IPs")
-	}
+	if locationsFlag {
+		ipBytes, err := object.GetObjectIPs(ctx, uplink.Config{}, pr.access, pr.bucket, o.Key)
```

Comment:
Can we avoid making additional requests to the satellite?

Also, where's the main bottleneck with regards to the slowness?

Reply:
that's the only one request where we get all the IPs. Is there any other way to get them?

main problem is that large file's pieces are stored on a big amount of nodes. That's why we need to look up into max mind db to get locations for each node. So the main bottleneck is here I think 
https://github.com/storj/linksharing/blob/main/objectmap/mapper.go#L100

Reply:
> that's the only one request where we get all the IPs. Is there any other way to get them?

Making a second request to the same endpoint will make 2x calls to the satellite -- since it'll need to stat, list etc. to figure out what the path means.

Reply:
Looking up the values from a map shouldn't take any significant amount of time. If it does, then there's something very wrong with the cache.